solution added

Vosker opened the pull request.

Vosker committed:
solution added

dmytro-hryhoruk commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
+
+        return width == null
+                ? width == rect.width : width.equals(width)
+                && (length == null || rect.length == null)
```
can we do here the same check as with width?

dmytro-hryhoruk commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
+    @Override
+    public int hashCode() {
+        int result = 61;
+        result = 31 * result + (getLength() == null ? 0 : getLength().hashCode());
+        result = 31 * result + (getWidth() == null ? 0 : getWidth().hashCode());
+        result = 31 * result + (getColor() == null ? 0 : getColor().hashCode());
```
should we use getters inside class?

Vosker committed:
Update Rectangle.java

Привет у меня последние полчаса проблема с тем чтобы запушить изменения на гит. Не знаю как решить проблему, сейчас гуглю. Но мне нужно залить домашку готовую до 5, а я не могу нормально залить ее. Поменял коммит прямо на сайте

dmytro-hryhoruk commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
+        }
+        Rectangle rect = (Rectangle) obj;
+
+        return (length != null && rect.length != null)
```
do we need to check both length on null?

Vosker committed:
Update Rectangle.java

dmytro-hryhoruk reviewed (commented):
fix travis

Vosker committed:
Update Rectangle.java
